connectReactNativeChat/src/components/ChatWrapper.js

```diff
@@ -0,0 +1,129 @@
+import NetInfo from '@react-native-community/netinfo'
+
+/**
+ * `amazon-connect-websocket-manager.js` depencency will use `navigator.onLine`
+ * Unsupported or mobile runtime will return `null/undefined` - preventing websocket connections
+ * Legacy browsers will always return `true` [ref: caniuse.com/netinfo]
+ */
+let isOnline = true
+const customIsNetworkOnline = () => isOnline
+
+/*
+ * Wrapper to manage ChatJS Session and pass messages down to GiftedChat component
+ * Handles state changes when loading/connecting
+ * Handles event listeners to update transcript
+ */
+const ChatWrapper = ({ navigation, ChatWidgetComponent }) => {
+  const [connected, setConnected] = useState(false)
+  const [loading, setLoading] = useState(false)
+  const [session, setSession] = useState(null)
+  const [messages, setMessages] = useState([])
+
+  const openHomeScreen = () => navigation.navigate('Home')
+
+  /**
+   * Network event listener native to device
+   * Will update `isOnline` value asynchronously whenever network calls are made
+   */
+  const unsubscribeNetworkEventListener = NetInfo.addEventListener((state) => {
+    isOnline = state.isConnected
+  })
+
+  useEffect(() => {
+    return unsubscribeNetworkEventListener()
+  }, [])
+
+  useEffect(() => {
+    if (!loading && !connected) {
+      submitChatInitiation()
+    }
+  }, [])
+
+  const handleStartChatFailure = (e) => {
+    console.error(e)
+    setConnected(false)
+    setLoading(false)
+    alert('StartChat request failed :(')
+    openHomeScreen()
+  }
+
+  const handleStartChatSuccess = () => {
+    setLoading(false)
+    setConnected(true)
+  }
+
+  const submitChatInitiation = async () => {
+    // Create chat session connection
+    setLoading(true)
+    const chatDetails = await initiateChat(startChatRequestInput, handleStartChatFailure)
+    const chatSession = new ChatSession(
+      chatDetails,
+      startChatRequestInput.name,
+      startChatRequestInput.region,
+      customIsNetworkOnline // pass down the custom "isNetworkOnline" function
+    )
+    await chatSession.openChatSession().then(handleStartChatSuccess).catch(handleStartChatFailure)
+    setSession(chatSession)
+
+    // Add event listeners to chat session
+    chatSession.onIncoming(async () => {
+      const latestTranscript = await chatSession.loadPreviousTranscript()
+      setMessages(filterIncomingMessages(latestTranscript))
+    })
+    chatSession.onOutgoing(async () => {
+      const latestTranscript = await chatSession.loadPreviousTranscript()
+      setMessages(filterIncomingMessages(latestTranscript))
+    })
+  }
+
+  const handleDisconnect = () => {
+    if (!loading && connected) {
+      setLoading(true)
+      session.endChat(() => {
+        setConnected(false)
+        setLoading(false)
+        openHomeScreen()
+      })
+    } else {
+      openHomeScreen()
+    }
+  }
+
+  const handleSendMessage = (msg) => {
+    const content = {
+      data: msg,
+      type: 'text/plain',
```

Comment:
Does this mean all the messages will be sent as plain text? Does markdown message work with this implementation?

Reply:
This example always renders plain text string, wether it is `"text/plain"` or `"text/markdown"`

Because CCP always sends `text/markdown` , and attempt to implement rich formatting

Reply:
Not implementing rich formatting to keep the example simple. Updating the `config.js` file to match